src/ray/object_manager/pull_manager.cc

```diff
@@ -0,0 +1,132 @@
+#include "ray/object_manager/pull_manager.h"
+
+namespace ray {
+
+PullManager::PullManager(
+    NodeID &self_node_id, const std::function<bool(const ObjectID &)> object_is_local,
+    const std::function<void(const ObjectID &, const NodeID &)> send_pull_request,
+    const std::function<int(int)> get_rand_int,
```

Comment:
Does this need to be a dependency? Can we just call the random number generator directly from this class?

Reply:
I think this just has the potential to make future unit testing easier easier (if we can control what nodes we pull from), but I don't have a strong preference and can change it if you feel strongly about it.

Reply:
I'd prefer we remove it since it's not clear yet what the future implementation and therefore testing should be.